File: user-service/src/main/java/rs/raf/user_service/domain/dto/AgentDto.java
package rs.raf.user_service.domain.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.math.BigDecimal;
import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AgentDto {

    private Long id;
    private String username;
    private String position;
    private String department;
    private boolean active;

    //from BaseUser
    private String firstName;
    private String lastName;
    private String email;
    private String jmbg;
    private Date birthDate;
    private String gender;
    private String phone;
    private String address;
    private String role;

    private BigDecimal limitAmount;
    private BigDecimal usedLimit;
    private boolean needsApproval;

}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/dto/employee/AgentDto.kt
package pack.userservicekotlin.domain.dto.employee

import java.math.BigDecimal
import java.util.*

class AgentDto(
    var id: Long?,
    val username: String?,
    val position: String?,
    val department: String?,
    val active: Boolean?,
    // from BaseUser
    val firstName: String?,
    val lastName: String?,
    val email: String?,
    val jmbg: String?,
    val birthDate: Date?,
    val gender: String?,
    val phone: String?,
    val address: String?,
    val role: String?,
    val limitAmount: BigDecimal,
    val usedLimit: BigDecimal,
    val needsApproval: Boolean,
)
